### ISSUE TITLE
World save folder not found

### ISSUE THREAD

**Issue by Gonaeng** [escapingnetwork/core-keeper-dedicated#66]
I tried all the solutions listed in the issue section, but the folder is not created.

![11](https://github.com/user-attachments/assets/4f319f08-1169-42ea-8f89-550e22e7d76c)
![22](https://github.com/user-attachments/assets/e14d5b57-013d-49e3-bd3d-a5091adfd54a)
![33](https://github.com/user-attachments/assets/671ea034-6180-4ca3-ac7c-f723e6259f72)
![55](https://github.com/user-attachments/assets/0a143022-5f49-49b7-8e86-1646f3e8f0f8)

**Comment by Gonaeng** [comment]
> Hm, can you share the logs too ? Also, set SEASON="", theres an issue if season is present in any way, it will override current irl season.

Mono path[0] = '/home/steam/core-keeper-dedicated/CoreKeeperServer_Data/Managed'
Mono config path = '/home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/etc'
Preloaded 'discord_game_sdk.so'
Shutdown handler: initialize.
Preloaded 'libEOSSDK-Linux-Shipping.so'
Preloaded 'libPlayFabMultiplayer.so'
Preloaded 'lib_burst_generated.so'
Preloaded 'libsentry.so'
Preloaded 'libsteam_api.so'
Unable to preload the following plugins:
	libparty.so
Display 0 'screen': 1x1 (primary device).
[Physics::Module] Initialized MultithreadedJobDispatcher with 1 workers.
Loading player data from /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/data.unity3d
Desktop is 1 x 1 @ 0 Hz
Initialize engine version: 2022.3.20f1 (61c2feb0970d)
[Subsystems] Discovering subsystems at path /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/UnitySubsystems
GfxDevice: creating device client; threaded=0; jobified=0
Renderer: llvmpipe (LLVM 15.0.6, 128 bits)
Vendor:   Mesa/X.org
Version:  4.5 (Core Profile) Mesa 22.3.6
GLES:     0
 GL_3DFX_texture_compression_FXT1 GL_AMD_conservative_depth GL_AMD_draw_buffers_blend GL_AMD_gpu_shader_int64 GL_AMD_multi_draw_indirect GL_AMD_pinned_memory GL_AMD_query_buffer_object GL_AMD_seamless_cubemap_per_texture GL_AMD_shader_stencil_export GL_AMD_shader_trinary_minmax GL_AMD_texture_texture4 GL_AMD_vertex_shader_layer GL_AMD_vertex_shader_viewport_index GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_ES2_compatibility GL_ARB_ES3_1_compatibility GL_ARB_ES3_2_compatibility GL_ARB_ES3_compatibility GL_ARB_arrays_of_arrays GL_ARB_base_instance GL_ARB_blend_func_extended GL_ARB_buffer_storage GL_ARB_clear_buffer_object GL_ARB_clear_texture GL_ARB_clip_control GL_ARB_compressed_texture_pixel_storage GL_ARB_compute_shader GL_ARB_conditional_render_inverted GL_ARB_conservative_depth GL_ARB_copy_buffer GL_ARB_copy_image GL_ARB_cull_distance GL_ARB_debug_output GL_ARB_depth_buffer_float GL_ARB_depth_clamp GL_ARB_derivative_control GL_ARB_direct_state_access GL_ARB_draw_buffers GL_A
RB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_draw_indirect GL_ARB_draw_instanced GL_ARB_enhanced_layouts GL_ARB_explicit_attrib_location GL_ARB_explicit_uniform_location GL_ARB_fragment_coord_conventions GL_ARB_fragment_layer_viewport GL_ARB_fragment_shader GL_ARB_framebuffer_no_attachments GL_ARB_framebuffer_object GL_ARB_framebuffer_sRGB GL_ARB_get_program_binary GL_ARB_get_texture_sub_image GL_ARB_gl_spirv GL_ARB_gpu_shader5 GL_ARB_gpu_shader_fp64 GL_ARB_gpu_shader_int64 GL_ARB_half_float_pixel GL_ARB_half_float_vertex GL_ARB_indirect_parameters GL_ARB_instanced_arrays GL_ARB_internalformat_query GL_ARB_internalformat_query2 GL_ARB_invalidate_subdata GL_ARB_map_buffer_alignment GL_ARB_map_buffer_range GL_ARB_multi_bind GL_ARB_multi_draw_indirect GL_ARB_occlusion_query2 GL_ARB_parallel_shader_compile GL_ARB_pipeline_statistics_query GL_ARB_pixel_buffer_object GL_ARB_point_sprite GL_ARB_polygon_offset_clamp GL_ARB_post_depth_coverage GL_ARB_program_interface_query GL_ARB_provoking_vertex GL
_ARB_query_buffer_object GL_ARB_robust_buffer_access_behavior GL_ARB_robustness GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_seamless_cubemap_per_texture GL_ARB_separate_shader_objects GL_ARB_shader_atomic_counter_ops GL_ARB_shader_atomic_counters GL_ARB_shader_ballot GL_ARB_shader_bit_encoding GL_ARB_shader_clock GL_ARB_shader_draw_parameters GL_ARB_shader_group_vote GL_ARB_shader_image_load_store GL_ARB_shader_image_size GL_ARB_shader_objects GL_ARB_shader_precision GL_ARB_shader_stencil_export GL_ARB_shader_storage_buffer_object GL_ARB_shader_subroutine GL_ARB_shader_texture_image_samples GL_ARB_shader_texture_lod GL_ARB_shader_viewport_layer_array GL_ARB_shading_language_420pack GL_ARB_shading_language_include GL_ARB_shading_language_packing GL_ARB_spirv_extensions GL_ARB_stencil_texturing GL_ARB_sync GL_ARB_tessellation_shader GL_ARB_texture_barrier GL_ARB_texture_buffer_object GL_ARB_texture_buffer_object_rgb32 GL_ARB_texture_buffer_range GL_ARB_texture_compression_bptc 
GL_ARB_texture_compression_rgtc GL_ARB_texture_cube_map_array GL_ARB_texture_filter_anisotropic GL_ARB_texture_filter_minmax GL_ARB_texture_float GL_ARB_texture_gather GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_multisample GL_ARB_texture_non_power_of_two GL_ARB_texture_query_levels GL_ARB_texture_query_lod GL_ARB_texture_rectangle GL_ARB_texture_rg GL_ARB_texture_rgb10_a2ui GL_ARB_texture_stencil8 GL_ARB_texture_storage GL_ARB_texture_storage_multisample GL_ARB_texture_swizzle GL_ARB_texture_view GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_transform_feedback_instanced GL_ARB_transform_feedback_overflow_query GL_ARB_uniform_buffer_object GL_ARB_vertex_array_bgra GL_ARB_vertex_array_object GL_ARB_vertex_attrib_64bit GL_ARB_vertex_attrib_binding GL_ARB_vertex_buffer_object GL_ARB_vertex_shader GL_ARB_vertex_type_10f_11f_11f_rev GL_ARB_vertex_type_2_10_10_10_rev GL_ARB_viewport_array GL_ARM_shader_framebuffer_fetch_depth_stencil GL_ATI_blend_equation_separate GL_ATI_te
xture_float GL_ATI_texture_mirror_once GL_EXT_EGL_image_storage GL_EXT_EGL_sync GL_EXT_abgr GL_EXT_blend_equation_separate GL_EXT_debug_label GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_framebuffer_object GL_EXT_framebuffer_sRGB GL_EXT_memory_object GL_EXT_memory_object_fd GL_EXT_packed_depth_stencil GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_polygon_offset_clamp GL_EXT_provoking_vertex GL_EXT_shader_framebuffer_fetch GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_shader_integer_mix GL_EXT_texture_array GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_compression_s3tc GL_EXT_texture_filter_anisotropic GL_EXT_texture_filter_minmax GL_EXT_texture_integer GL_EXT_texture_mirror_clamp GL_EXT_texture_sRGB GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_RG8 GL_EXT_texture_sRGB_decode GL_EXT_texture_shadow_lod GL_EXT_texture_shared_exponent GL_EXT_texture_snorm GL_EXT_textu
re_swizzle GL_EXT_timer_query GL_EXT_transform_feedback GL_EXT_vertex_array_bgra GL_EXT_vertex_attrib_64bit GL_IBM_multimode_draw_arrays GL_INTEL_shader_atomic_float_minmax GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_debug GL_KHR_no_error GL_KHR_parallel_shader_compile GL_KHR_robust_buffer_access_behavior GL_KHR_robustness GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_sliced_3d GL_MESA_framebuffer_flip_y GL_MESA_pack_invert GL_MESA_shader_integer_functions GL_MESA_texture_signed_rgba GL_MESA_ycbcr_texture GL_NV_conditional_render GL_NV_copy_image GL_NV_depth_clamp GL_NV_packed_depth_stencil GL_NV_shader_atomic_float GL_NV_texture_barrier GL_OES_EGL_image GL_S3_s3tc
OPENGL LOG: Creating OpenGL 4.5 graphics device ; Context level  <OpenGL 4.5> ; Context handle 13438336
Begin MonoManager ReloadAssembly
- Loaded All Assemblies, in  0.298 seconds
- Finished resetting the current domain, in  0.004 seconds
PlatformConfiguration: loading platform configuration for variant Linux.
PlatformConfiguration: platform configuration for variant Linux was not found.
PlatformConfiguration: loading platform configuration for variant PC.
Fallback handler could not load library /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/x86_64/data-0x4cc5410.so
Fallback handler could not load library /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/x86_64/data-0x4e60600.so
Initializing Steamworks with AppID 1621690
[S_API] SteamAPI_Init(): Loaded local 'steamclient.so' OK.
IPC function call IClientUtils::GetSteamRealm took too long: 43 msec
Setting breakpad minidump AppID = 1621690
[S_API FAIL] Tried to access Steam interface SteamNetworkingUtils004 before SteamAPI_Init succeeded.
Steam API initialized
Logged in with userid:3228921874
UnloadTime: 4.184110 ms
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Checking system requirements: PASS
Manager prelaunch and GO initialization starting...
Initialize global manager
Global manager Resources.Load
Start initializing all managers
Core count is 2
Job worker count is 1
PlatformConfiguration: applying settings for platform variant Linux.
PlatformConfiguration: applying PrefsManager visual quality settings.
PlatformConfiguration: applying PrefsManager audio settings.
PlatformConfiguration: applying PrefsManager session settings.
PlatformConfiguration: applying QualitySettings.
Version: 1.0.0.14-34ff
Time (UTC): 2024-10-29T13:16:28Z
Skipping faction mod /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/StreamingAssets/Conf/Factions/None.json with faction None (0)
Skipping loot mod /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/StreamingAssets/Conf/Loot/Empty.json for loot table Empty (0)
Found 319 converter(s)
Added post converter DropLootPostConverter
Added post converter DealDamagePostConverter
Added post converter RotationPostConverter
Added post converter BoatRidingColliderVariationPostConverter
Added post converter ReceivePushbackPostConverter
Added post converter ModPostConverter
Added post converter PugConversion.GhostPostConverter
Added post converter DisablePhysicsRestoreConverter
ECSManager.Init: converting PugDatabase prefab list.
ECSManager.Init: converting mods extra authoring.
ECSManager.Init: converting server authoring prefab.
PugDatabase initialized 2357 prefabs
FinishConversionQueue: forcing a frame update at 500.105 ms from previous.
ECSManager.Init: post conversion for server authoring prefab.
RunPostConverters: forcing a frame update at 503.219 ms from previous.
RunPostConverters: forcing a frame update at 500.022 ms from previous.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on Unity.Entities.CompanionGameObjectUpdateTransformSystem targeting Unity.Transforms.TransformSystemGroup.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Forcing season -1
StartGameRPCSystem OnCreate
No mods have registered for ObjectCreated or ObjectDestroyed callbacks. Disabling ObjectCreatedSystem.
ECSManager: removing worlds from the update loop.
ECSManager: converting authoring data.
ECSManager: converting authoring data for the server world.
Found 319 converter(s)
Added post converter DropLootPostConverter
Added post converter DealDamagePostConverter
Added post converter RotationPostConverter
Added post converter BoatRidingColliderVariationPostConverter
Added post converter ReceivePushbackPostConverter
Added post converter ModPostConverter
Added post converter PugConversion.GhostPostConverter
Added post converter DisablePhysicsRestoreConverter
PugDatabase initialized 2357 prefabs
FinishConversionQueue: forcing a frame update at 500.069 ms from previous.
FinishConversionQueue: forcing a frame update at 500.269 ms from previous.
RunPostConverters: forcing a frame update at 500.375 ms from previous.
RunPostConverters: forcing a frame update at 500.135 ms from previous.
RunPostConverters: forcing a frame update at 500.346 ms from previous.
ECSManager: converting authoring data for the server world complete.
ECSManager: adding worlds to the update loop.
ECSManager: initializing worlds.
Simulation tick rate=20 Network tick rate=20
disable default variant system group for world ServerWorld
disable CompanionGameObjectUpdateTransformSystem for world ServerWorld
disable ParentSystem for world ServerWorld
Listening on SteamID userid:90207300131057682
ConvertAuthoringDataRoutine complete. Process was successful.
number prefabs 2342
Started session with Game ID zSVDbmJgkrTShsuiRDp28VjHEqC3
no existing save file, starting with empty world
starting a new world : a710aaa4644273abe5584198f84e6738
Hashed string seed 0 -> 4108050209
Using seed 4108050209
Starting new world with generation type FullRelease
Adding marker for LarvaHiveBossDungeon at int2(220, -201)
Adding marker for ShamanBossRoom at int2(222, 189)
Adding marker for BirdBoss at int2(276, 473)
Adding marker for PoisonSlimeBossDungeon at int2(495, 341)
Adding marker for SlipperySlimeBossDungeon at int2(-549, -455)
Adding marker for OctopusBossArea at int2(-430, -478)
Adding marker for ScarabBossArea at int2(-167, -568)
Adding marker for LavaSlimeBossDungeon at int2(486, -486)
Adding marker for CrystalMeteorScene at int2(14, -697)
timescale = 0


Here it is :)

**Comment by Diyagi** [comment]
It is possible that it wont save the world until someone joins the server, even if you close the server, so try to join the server, break something then close the server.

**Comment by Gonaeng** [comment]
> If this does not work, see if you can provide the logs after the server closes to see if theres something going wrong in the server quit sequence.

Mono path[0] = '/home/steam/core-keeper-dedicated/CoreKeeperServer_Data/Managed'
Mono config path = '/home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/etc'
Preloaded 'discord_game_sdk.so'
Shutdown handler: initialize.
Preloaded 'libEOSSDK-Linux-Shipping.so'
Preloaded 'libPlayFabMultiplayer.so'
Preloaded 'lib_burst_generated.so'
Preloaded 'libsentry.so'
Preloaded 'libsteam_api.so'
Unable to preload the following plugins:
	libparty.so
Display 0 'screen': 1x1 (primary device).
[Physics::Module] Initialized MultithreadedJobDispatcher with 1 workers.
Loading player data from /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/data.unity3d
Desktop is 1 x 1 @ 0 Hz
Initialize engine version: 2022.3.20f1 (61c2feb0970d)
[Subsystems] Discovering subsystems at path /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/UnitySubsystems
GfxDevice: creating device client; threaded=0; jobified=0
Renderer: llvmpipe (LLVM 15.0.6, 128 bits)
Vendor:   Mesa/X.org
Version:  4.5 (Core Profile) Mesa 22.3.6
GLES:     0
 GL_3DFX_texture_compression_FXT1 GL_AMD_conservative_depth GL_AMD_draw_buffers_blend GL_AMD_gpu_shader_int64 GL_AMD_multi_draw_indirect GL_AMD_pinned_memory GL_AMD_query_buffer_object GL_AMD_seamless_cubemap_per_texture GL_AMD_shader_stencil_export GL_AMD_shader_trinary_minmax GL_AMD_texture_texture4 GL_AMD_vertex_shader_layer GL_AMD_vertex_shader_viewport_index GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_ES2_compatibility GL_ARB_ES3_1_compatibility GL_ARB_ES3_2_compatibility GL_ARB_ES3_compatibility GL_ARB_arrays_of_arrays GL_ARB_base_instance GL_ARB_blend_func_extended GL_ARB_buffer_storage GL_ARB_clear_buffer_object GL_ARB_clear_texture GL_ARB_clip_control GL_ARB_compressed_texture_pixel_storage GL_ARB_compute_shader GL_ARB_conditional_render_inverted GL_ARB_conservative_depth GL_ARB_copy_buffer GL_ARB_copy_image GL_ARB_cull_distance GL_ARB_debug_output GL_ARB_depth_buffer_float GL_ARB_depth_clamp GL_ARB_derivative_control GL_ARB_direct_state_access GL_ARB_draw_buffers GL_A
RB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_draw_indirect GL_ARB_draw_instanced GL_ARB_enhanced_layouts GL_ARB_explicit_attrib_location GL_ARB_explicit_uniform_location GL_ARB_fragment_coord_conventions GL_ARB_fragment_layer_viewport GL_ARB_fragment_shader GL_ARB_framebuffer_no_attachments GL_ARB_framebuffer_object GL_ARB_framebuffer_sRGB GL_ARB_get_program_binary GL_ARB_get_texture_sub_image GL_ARB_gl_spirv GL_ARB_gpu_shader5 GL_ARB_gpu_shader_fp64 GL_ARB_gpu_shader_int64 GL_ARB_half_float_pixel GL_ARB_half_float_vertex GL_ARB_indirect_parameters GL_ARB_instanced_arrays GL_ARB_internalformat_query GL_ARB_internalformat_query2 GL_ARB_invalidate_subdata GL_ARB_map_buffer_alignment GL_ARB_map_buffer_range GL_ARB_multi_bind GL_ARB_multi_draw_indirect GL_ARB_occlusion_query2 GL_ARB_parallel_shader_compile GL_ARB_pipeline_statistics_query GL_ARB_pixel_buffer_object GL_ARB_point_sprite GL_ARB_polygon_offset_clamp GL_ARB_post_depth_coverage GL_ARB_program_interface_query GL_ARB_provoking_vertex GL
_ARB_query_buffer_object GL_ARB_robust_buffer_access_behavior GL_ARB_robustness GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_seamless_cubemap_per_texture GL_ARB_separate_shader_objects GL_ARB_shader_atomic_counter_ops GL_ARB_shader_atomic_counters GL_ARB_shader_ballot GL_ARB_shader_bit_encoding GL_ARB_shader_clock GL_ARB_shader_draw_parameters GL_ARB_shader_group_vote GL_ARB_shader_image_load_store GL_ARB_shader_image_size GL_ARB_shader_objects GL_ARB_shader_precision GL_ARB_shader_stencil_export GL_ARB_shader_storage_buffer_object GL_ARB_shader_subroutine GL_ARB_shader_texture_image_samples GL_ARB_shader_texture_lod GL_ARB_shader_viewport_layer_array GL_ARB_shading_language_420pack GL_ARB_shading_language_include GL_ARB_shading_language_packing GL_ARB_spirv_extensions GL_ARB_stencil_texturing GL_ARB_sync GL_ARB_tessellation_shader GL_ARB_texture_barrier GL_ARB_texture_buffer_object GL_ARB_texture_buffer_object_rgb32 GL_ARB_texture_buffer_range GL_ARB_texture_compression_bptc 
GL_ARB_texture_compression_rgtc GL_ARB_texture_cube_map_array GL_ARB_texture_filter_anisotropic GL_ARB_texture_filter_minmax GL_ARB_texture_float GL_ARB_texture_gather GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_multisample GL_ARB_texture_non_power_of_two GL_ARB_texture_query_levels GL_ARB_texture_query_lod GL_ARB_texture_rectangle GL_ARB_texture_rg GL_ARB_texture_rgb10_a2ui GL_ARB_texture_stencil8 GL_ARB_texture_storage GL_ARB_texture_storage_multisample GL_ARB_texture_swizzle GL_ARB_texture_view GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_transform_feedback_instanced GL_ARB_transform_feedback_overflow_query GL_ARB_uniform_buffer_object GL_ARB_vertex_array_bgra GL_ARB_vertex_array_object GL_ARB_vertex_attrib_64bit GL_ARB_vertex_attrib_binding GL_ARB_vertex_buffer_object GL_ARB_vertex_shader GL_ARB_vertex_type_10f_11f_11f_rev GL_ARB_vertex_type_2_10_10_10_rev GL_ARB_viewport_array GL_ARM_shader_framebuffer_fetch_depth_stencil GL_ATI_blend_equation_separate GL_ATI_te
xture_float GL_ATI_texture_mirror_once GL_EXT_EGL_image_storage GL_EXT_EGL_sync GL_EXT_abgr GL_EXT_blend_equation_separate GL_EXT_debug_label GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_framebuffer_object GL_EXT_framebuffer_sRGB GL_EXT_memory_object GL_EXT_memory_object_fd GL_EXT_packed_depth_stencil GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_polygon_offset_clamp GL_EXT_provoking_vertex GL_EXT_shader_framebuffer_fetch GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_shader_integer_mix GL_EXT_texture_array GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_compression_s3tc GL_EXT_texture_filter_anisotropic GL_EXT_texture_filter_minmax GL_EXT_texture_integer GL_EXT_texture_mirror_clamp GL_EXT_texture_sRGB GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_RG8 GL_EXT_texture_sRGB_decode GL_EXT_texture_shadow_lod GL_EXT_texture_shared_exponent GL_EXT_texture_snorm GL_EXT_textu
re_swizzle GL_EXT_timer_query GL_EXT_transform_feedback GL_EXT_vertex_array_bgra GL_EXT_vertex_attrib_64bit GL_IBM_multimode_draw_arrays GL_INTEL_shader_atomic_float_minmax GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_debug GL_KHR_no_error GL_KHR_parallel_shader_compile GL_KHR_robust_buffer_access_behavior GL_KHR_robustness GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_sliced_3d GL_MESA_framebuffer_flip_y GL_MESA_pack_invert GL_MESA_shader_integer_functions GL_MESA_texture_signed_rgba GL_MESA_ycbcr_texture GL_NV_conditional_render GL_NV_copy_image GL_NV_depth_clamp GL_NV_packed_depth_stencil GL_NV_shader_atomic_float GL_NV_texture_barrier GL_OES_EGL_image GL_S3_s3tc
OPENGL LOG: Creating OpenGL 4.5 graphics device ; Context level  <OpenGL 4.5> ; Context handle 19462464
Begin MonoManager ReloadAssembly
- Loaded All Assemblies, in  0.304 seconds
- Finished resetting the current domain, in  0.004 seconds
PlatformConfiguration: loading platform configuration for variant Linux.
PlatformConfiguration: platform configuration for variant Linux was not found.
PlatformConfiguration: loading platform configuration for variant PC.
Fallback handler could not load library /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/x86_64/data-0x58e6bc0.so
Fallback handler could not load library /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/MonoBleedingEdge/x86_64/data-0x5ad0ef0.so
Initializing Steamworks with AppID 1621690
[S_API] SteamAPI_Init(): Loaded local 'steamclient.so' OK.
Setting breakpad minidump AppID = 1621690
[S_API FAIL] Tried to access Steam interface SteamNetworkingUtils004 before SteamAPI_Init succeeded.
Steam API initialized
Logged in with userid:2535600146
UnloadTime: 4.120110 ms
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Checking system requirements: PASS
Manager prelaunch and GO initialization starting...
Initialize global manager
Global manager Resources.Load
Start initializing all managers
Core count is 2
Job worker count is 1
PlatformConfiguration: applying settings for platform variant Linux.
PlatformConfiguration: applying PrefsManager visual quality settings.
PlatformConfiguration: applying PrefsManager audio settings.
PlatformConfiguration: applying PrefsManager session settings.
PlatformConfiguration: applying QualitySettings.
Version: 1.0.0.14-34ff
Time (UTC): 2024-10-29T15:43:06Z
Skipping faction mod /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/StreamingAssets/Conf/Factions/None.json with faction None (0)
Skipping loot mod /home/steam/core-keeper-dedicated/CoreKeeperServer_Data/StreamingAssets/Conf/Loot/Empty.json for loot table Empty (0)
Found 319 converter(s)
Added post converter DropLootPostConverter
Added post converter DealDamagePostConverter
Added post converter RotationPostConverter
Added post converter BoatRidingColliderVariationPostConverter
Added post converter ReceivePushbackPostConverter
Added post converter ModPostConverter
Added post converter PugConversion.GhostPostConverter
Added post converter DisablePhysicsRestoreConverter
ECSManager.Init: converting PugDatabase prefab list.
ECSManager.Init: converting mods extra authoring.
ECSManager.Init: converting server authoring prefab.
PugDatabase initialized 2357 prefabs
FinishConversionQueue: forcing a frame update at 500.328 ms from previous.
ECSManager.Init: post conversion for server authoring prefab.
RunPostConverters: forcing a frame update at 500.004 ms from previous.
RunPostConverters: forcing a frame update at 500.544 ms from previous.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on Unity.Entities.CompanionGameObjectUpdateTransformSystem targeting Unity.Transforms.TransformSystemGroup.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateAfterAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.GhostSendSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on InitializaAttackPlayerSpawnedGhostSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
Ignoring invalid [Unity.Entities.UpdateBeforeAttribute] attribute on WorkaroundForRpcSystemNotDependentOnRpcCommandRequestersSystem targeting Unity.NetCode.RpcSystem.
This attribute can only order systems that are members of the same ComponentSystemGroup instance.
Make sure that both systems are in the same system group with [UpdateInGroup(typeof(Unity.Entities.SimulationSystemGroup))],
or by manually adding both systems to the same group's update list.
New season: Halloween
StartGameRPCSystem OnCreate
No mods have registered for ObjectCreated or ObjectDestroyed callbacks. Disabling ObjectCreatedSystem.
ECSManager: removing worlds from the update loop.
ECSManager: converting authoring data.
ECSManager: converting authoring data for the server world.
Found 319 converter(s)
Added post converter DropLootPostConverter
Added post converter DealDamagePostConverter
Added post converter RotationPostConverter
Added post converter BoatRidingColliderVariationPostConverter
Added post converter ReceivePushbackPostConverter
Added post converter ModPostConverter
Added post converter PugConversion.GhostPostConverter
Added post converter DisablePhysicsRestoreConverter
PugDatabase initialized 2357 prefabs
FinishConversionQueue: forcing a frame update at 500.527 ms from previous.
FinishConversionQueue: forcing a frame update at 502.138 ms from previous.
RunPostConverters: forcing a frame update at 500.051 ms from previous.
RunPostConverters: forcing a frame update at 500.048 ms from previous.
RunPostConverters: forcing a frame update at 500.144 ms from previous.
ECSManager: converting authoring data for the server world complete.
ECSManager: adding worlds to the update loop.
ECSManager: initializing worlds.
Simulation tick rate=20 Network tick rate=20
disable default variant system group for world ServerWorld
disable CompanionGameObjectUpdateTransformSystem for world ServerWorld
disable ParentSystem for world ServerWorld
Listening on SteamID userid:90207303732703250
ConvertAuthoringDataRoutine complete. Process was successful.
number prefabs 2342
Started session with Game ID bvpvPP6a3VjMucnq5Cu8HifZQ73C
no existing save file, starting with empty world
starting a new world : 2d3b4526a4e6079b127b6a1aa6d79c6c
Hashed string seed 0 -> 4108050209
Using seed 4108050209
Starting new world with generation type FullRelease
Adding marker for LarvaHiveBossDungeon at int2(220, -201)
Adding marker for ShamanBossRoom at int2(222, 189)
Adding marker for BirdBoss at int2(276, 473)
Adding marker for PoisonSlimeBossDungeon at int2(495, 341)
Adding marker for SlipperySlimeBossDungeon at int2(-549, -455)
Adding marker for OctopusBossArea at int2(-430, -478)
Adding marker for ScarabBossArea at int2(-167, -568)
Adding marker for LavaSlimeBossDungeon at int2(486, -486)
Adding marker for CrystalMeteorScene at int2(14, -697)
timescale = 0
Accepted connection from 76561198056526695 with result OK awaiting authentication
SteamNet Warning: [#2320726969 P2P steamid:76561198056526695 vport 0] Relay candidates enabled by P2P_Transport_ICE_Enable, but P2P_TURN_ServerList is empty
Connected to userid:76561198056526695
Successful authentication from userid:76561198056526695
timescale = 1
Read cartography table map with 0 parts
Running 0 electricity updates initially
server has ghost collection hash 11858044417851913081 for 1828 ghosts
Accepted connect RPC
spawning player with character data size 16682
Creating new player entity
[userid:76561198056526695] player Gonaeng connected islocalplayer=False
[userid:76561198056526695] is using new name Gonaeng
Started game for new connection
Spawning StartRoom
Spawning SlimeBoss
Spawning LarvaBossSpawn
Spawning CavelingMerchantRoom
Spawning FishingMerchantArea
[userid:76561198056526695] add or refresh condition None 0 duration 0 for Player: amount=1, variation=0
Spawning BirdBoss
Removing fake marker for BirdBoss
Disconnected from userid:76561198056526695 with reason App_Min
check connection state
player disconnect
timescale = 0
Got quit request
Exit blocked by ECSManager
Quit blocked
timescale = 1
Got quit request
Save is still pending
Exit blocked by ECSManager
Quit blocked


If you close and turn on the server, the world will be reset. :(

**Comment by Diyagi** [comment]
Oh, the server is not shutting down properly, this is what the shutdown should look like

```
timescale = 0
Got quit request
Exit blocked by ECSManager
Quit blocked
timescale = 1
Got quit request
Save is still pending
Exit blocked by ECSManager
Quit blocked
Read cartography table map with 0 parts
Running 0 electricity updates initially
Allocating a new file write buffer of size 8.000MB for WorldSave. Total buffer size for this file including pool is 8.000MB
timescale = 0
Got quit request
Running quit handlers
Setting up 1 worker threads for Enlighten.
PlatformManager was destroyed
Memory Statistics:
[ALLOC_TEMP_TLS] TLS Allocator
  StackAllocators :
    [ALLOC_TEMP_MAIN]
      Peak usage frame count: [0-1.0 KB]: 415 frames, [4.0 KB-8.0 KB]: 1 frames, [8.0 KB-16.0 KB]: 79 frames, [32.0 KB-64.0 KB]: 1 frames, [64.0 KB-128.0 KB]: 2 frames, [1.0 MB-2.0 MB]: 1 frames, [4.0 MB-8.0 MB]: 1 frames
      Initial Block Size 128.0 MB
      Current Block Size 128.0 MB
      Peak Allocated Bytes 7.9 MB
      Overflow Count 18
    [ALLOC_TEMP_Background Job.worker 4]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 0 B
      Overflow Count 0
    [ALLOC_TEMP_Loading.PreloadManager]
      Initial Block Size 256.0 KB
      Current Block Size 376.0 KB
      Peak Allocated Bytes 372.1 KB
      Overflow Count 4
    [ALLOC_TEMP_Background Job.worker 3]
      Initial Block Size 32.0 KB
      Current Block Size 32.0 KB
      Peak Allocated Bytes 0 B
      Overflow Count 0
//...//
[ALLOC_TYPETREE] Dual Thread Allocator
  Peak main deferred allocation count 0
    [ALLOC_BUCKET]
      Large Block size 4.0 MB
      Used Block count 1
      Peak Allocated bytes 4.0 MB
      Failed Allocations. Bucket layout:
        16B: 31 Subsections = 31744 buckets. Failed count: 210288
        32B: 5 Subsections = 2560 buckets. Failed count: 140879
        48B: 15 Subsections = 5120 buckets. Failed count: 20657
        64B: 179 Subsections = 45824 buckets. Failed count: 551406
        80B: 19 Subsections = 3891 buckets. Failed count: 8259
        96B: 3 Subsections = 512 buckets. Failed count: 5698
        112B: 2 Subsections = 292 buckets. Failed count: 1685
        128B: 2 Subsections = 256 buckets. Failed count: 91637
    [ALLOC_TYPETREE_MAIN]
      Peak usage frame count: [128.0 KB-256.0 KB]: 500 frames
      Requested Block Size 2.0 MB
      Peak Block count 1
      Peak Allocated memory 181.0 KB
      Peak Large allocation bytes 0 B
    [ALLOC_TYPETREE_THREAD]
      Peak usage frame count: [4.0 KB-8.0 KB]: 500 frames
      Requested Block Size 2.0 MB
      Peak Block count 1
      Peak Allocated memory 4.5 KB
      Peak Large allocation bytes 0 B
Shutdown handler: cleanup.
```
Im not sure why tho, im assuming you did not overridden the entrypoint ?

**Comment by Diyagi** [comment]
Could you try the image from my PR (#62) ?
`diyagi/core-keeper-dedicated`
Theres some changes to the env vars
```env
PUID=1000
PGID=1000
WORLD_INDEX=0
WORLD_NAME="Core Keeper Server"
WORLD_SEED=0
WORLD_MODE=0
GAME_ID=""
DATA_PATH="${STEAMAPPDATADIR}"
MAX_PLAYERS=10
SEASON=""
SERVER_IP=""
SERVER_PORT=""
DISCORD_WEBHOOK_URL=""
# Player Join
DISCORD_PLAYER_JOIN_ENABLED=true
DISCORD_PLAYER_JOIN_MESSAGE="$${char_name} ($${steamid}) has joined the server."
DISCORD_PLAYER_JOIN_TITLE="Player Joined"
DISCORD_PLAYER_JOIN_COLOR="47456"
# Player Leave
DISCORD_PLAYER_LEAVE_ENABLED=true
DISCORD_PLAYER_LEAVE_MESSAGE="$${char_name} ($${steamid}) has disconnected. Reason: $${reason}."
DISCORD_PLAYER_LEAVE_TITLE="Player Left"
DISCORD_PLAYER_LEAVE_COLOR="11477760"
# Server Start
DISCORD_SERVER_START_ENABLED=true
DISCORD_SERVER_START_MESSAGE="**World:** $${world_name}\n**GameID:** $${gameid}"
DISCORD_SERVER_START_TITLE="Server Started"
DISCORD_SERVER_START_COLOR="2013440"
# Server Stop
DISCORD_SERVER_STOP_ENABLED=true
DISCORD_SERVER_STOP_MESSAGE=""
DISCORD_SERVER_STOP_TITLE="Server Stopped"
DISCORD_SERVER_STOP_COLOR="12779520"
```

**Comment by Diyagi** [comment]
> > Could you try the image from my PR (#62) ? `diyagi/core-keeper-dedicated` Theres some changes to the env vars
> > ```dotenv
> > PUID=1000
> > PGID=1000
> > WORLD_INDEX=0
> > WORLD_NAME="Core Keeper Server"
> > WORLD_SEED=0
> > WORLD_MODE=0
> > GAME_ID=""
> > DATA_PATH="${STEAMAPPDATADIR}"
> > MAX_PLAYERS=10
> > SEASON=""
> > SERVER_IP=""
> > SERVER_PORT=""
> > DISCORD_WEBHOOK_URL=""
> > # Player Join
> > DISCORD_PLAYER_JOIN_ENABLED=true
> > DISCORD_PLAYER_JOIN_MESSAGE="$${char_name} ($${steamid}) has joined the server."
> > DISCORD_PLAYER_JOIN_TITLE="Player Joined"
> > DISCORD_PLAYER_JOIN_COLOR="47456"
> > # Player Leave
> > DISCORD_PLAYER_LEAVE_ENABLED=true
> > DISCORD_PLAYER_LEAVE_MESSAGE="$${char_name} ($${steamid}) has disconnected. Reason: $${reason}."
> > DISCORD_PLAYER_LEAVE_TITLE="Player Left"
> > DISCORD_PLAYER_LEAVE_COLOR="11477760"
> > # Server Start
> > DISCORD_SERVER_START_ENABLED=true
> > DISCORD_SERVER_START_MESSAGE="**World:** $${world_name}\n**GameID:** $${gameid}"
> > DISCORD_SERVER_START_TITLE="Server Started"
> > DISCORD_SERVER_START_COLOR="2013440"
> > # Server Stop
> > DISCORD_SERVER_STOP_ENABLED=true
> > DISCORD_SERVER_STOP_MESSAGE=""
> > DISCORD_SERVER_STOP_TITLE="Server Stopped"
> > DISCORD_SERVER_STOP_COLOR="12779520"
> > ```
> 
> I'm sorry. I'm a beginner so I don't know how to apply this to my server.

These are env vars, you apply them with the `core.env` file, you dont need to have all of them in the files, just the ones you want to modify, if you are using an UI to deploy the container, theres probably already an field to put the env vars and its values (as i can see in the 4th image of the OP) so you may not need the `core.env` file.

If you want i can help more if you specify on what you are deploying the container.

**Comment by Gonaeng** [comment]
> > > Could you try the image from my PR (#62) ? `diyagi/core-keeper-dedicated` Theres some changes to the env vars
> > > ```dotenv
> > > PUID=1000
> > > PGID=1000
> > > WORLD_INDEX=0
> > > WORLD_NAME="Core Keeper Server"
> > > WORLD_SEED=0
> > > WORLD_MODE=0
> > > GAME_ID=""
> > > DATA_PATH="${STEAMAPPDATADIR}"
> > > MAX_PLAYERS=10
> > > SEASON=""
> > > SERVER_IP=""
> > > SERVER_PORT=""
> > > DISCORD_WEBHOOK_URL=""
> > > # Player Join
> > > DISCORD_PLAYER_JOIN_ENABLED=true
> > > DISCORD_PLAYER_JOIN_MESSAGE="$${char_name} ($${steamid}) has joined the server."
> > > DISCORD_PLAYER_JOIN_TITLE="Player Joined"
> > > DISCORD_PLAYER_JOIN_COLOR="47456"
> > > # Player Leave
> > > DISCORD_PLAYER_LEAVE_ENABLED=true
> > > DISCORD_PLAYER_LEAVE_MESSAGE="$${char_name} ($${steamid}) has disconnected. Reason: $${reason}."
> > > DISCORD_PLAYER_LEAVE_TITLE="Player Left"
> > > DISCORD_PLAYER_LEAVE_COLOR="11477760"
> > > # Server Start
> > > DISCORD_SERVER_START_ENABLED=true
> > > DISCORD_SERVER_START_MESSAGE="**World:** $${world_name}\n**GameID:** $${gameid}"
> > > DISCORD_SERVER_START_TITLE="Server Started"
> > > DISCORD_SERVER_START_COLOR="2013440"
> > > # Server Stop
> > > DISCORD_SERVER_STOP_ENABLED=true
> > > DISCORD_SERVER_STOP_MESSAGE=""
> > > DISCORD_SERVER_STOP_TITLE="Server Stopped"
> > > DISCORD_SERVER_STOP_COLOR="12779520"
> > > ```
> > 
> > 
> > I'm sorry. I'm a beginner so I don't know how to apply this to my server.
> 
> These are env vars, you apply them with the `core.env` file, you dont need to have all of them in the file, just the ones you want to modify, if you are using an UI to deploy the container, theres probably already an field to put the env vars and its values (as i can see in the 4th image of the OP) so you may not need the `core.env` file.
> 
> If you want i can help more if you specify on what you are deploying the container.

thx! it work!

**Comment by Diyagi** [comment]
@Gonaeng the changes from my image got merged in here, so remeber to swap back to `escaping/core-keeper-dedicated:latest`
